hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -359,4 +362,16 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tableState: HoodieTableState,
   protected def getLogFilesFromSlice(fileSlice: FileSlice): List[HoodieLogFile] = {
     fileSlice.getLogFiles.sorted(HoodieLogFile.getLogFileComparator).iterator().asScala.toList
   }
+
+  protected def makeMappingIterator(iter: HoodieFileGroupReader.HoodieFileGroupReaderIterator[InternalRow],
+                                    f: Function[InternalRow, InternalRow]): Iterator[InternalRow] = {
+    val mapIter = new CloseableMappingIterator[InternalRow, InternalRow](iter, JFunction.toJavaFunction(f))
```

Comment:
With this, we have three iterators. Is it possible for us to remove this intermediate iterator?